### PR TITLE
Response 404 for module product listings not found error

### PR DESCRIPTION
**Pull request by hlin**

### CODE REVIEW

File: product_listings_manager/products.py [hunk 1]
```diff
@@ -453,18 +453,15 @@ def getModuleProductListings(productLabel, moduleNVR):
...
         module_version = module['version']
-    except KeyError:
-        raise koji.GenericError("It's not a module build: %s" % moduleNVR)
+    except (KeyError, TypeError):
+        raise ProductListingsNotFoundError("It's not a module build: %s" % moduleNVR)
 
     prodinfo = Products.get_product_info(compose_dbh, productLabel)
-    if not prodinfo:
-        # no product with the given label exists
-        raise koji.GenericError("Could not find a product with label: %s" % productLabel)
```
Comment (hluk) [inline product_listings_manager/products.py]: @hlin Why was this removed? I think the line below would fail if the type `prodinfo` is wrong or it doesn't have exactly two items.

Comment (hlin) [reply product_listings_manager/products.py]: It's covered in get_product_info() https://github.com/release-engineering/product-listings-manager/blob/master/product_listings_manager/products.py#L82-L96

Comment (hluk) [reply product_listings_manager/products.py]: Oh, OK. Thanks for the new release!
